### PR TITLE
config: removing the legacy rest API

### LINKED ISSUE #5378: [v1.9.0 deprecation] Remove features marked deprecated in #4845
#4845 (config: deprecating legacy REST api) introduced a deprecation notice for v1.9.0. This issue tracks source code cleanup.

### PR DESCRIPTION
**Pull request by alyssawilk**
*Risk Level*: Medium (still unclear who might be using it)
*Testing*: n/a
*Docs Changes*: v1 docs already removed
*Release Notes*: inline
Fixes #5378

### CODE REVIEW

**Comment by alyssawilk** [comment]
So @envoyproxy/maintainers what do we think?

**Comment by alyssawilk** [comment]
Hah, that's what I get for doing a fresh master merge before upload.
Will sync and merge once master is fixed, but I did have everything working locally 5 minutes ago :-P

**Comment by mattklein123** [comment]
So much red. Amazing! Can you merge master?

**Comment by moderation** [comment]
Does this PR mean that https://github.com/lyft/discovery will cease to work? :cry:

**Comment by mattklein123** [comment]
@moderation yeah, though I think @taiki45 has a fork with v2 support.

**Comment by moderation** [comment]
Yes @taiki45 has https://github.com/cookpad/sds but I think there is a hard dependency on DynamoDB. (wait 10 mins). I've created a DynamoDB table and even posted a service registration. I have literally zero idea how `sds` is talking to DynamoDB but it is a start.

**Comment by taiki45** [comment]
@moderation We can help solving the issue, so could you file an issue to https://github.com/cookpad/sds?

File: docs/root/intro/version_history.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Version history
 * config: removed deprecated_v1 sds_config from :ref:`Bootstrap config <config_overview_v2_bootstrap>`.
 * cors: added :ref:`filter_enabled & shadow_enabled RuntimeFractionalPercent flags <cors-runtime>` to filter.
 * http: added new grpc_http1_reverse_bridge filter for converting gRPC requests into HTTP/1.1 requests.
+* config: removed REST_LEGACY as a valid :ref:`ApiType <envoy_api_field_core.ApiConfigSource.api_type>`.
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]: nit: alpha order

**Comment by alyssawilk** [comment]
Ok, here goes...

**Pull request merged**